Fix breadcrumbs when using a non-default hub

jeandeaual opened the pull request:
It works when explicitly creating a scope with `log.With(zapsentry.NewScope())`, but doing so loses the information tied to the hub (e.g. the `http.Request` info when using https://docs.sentry.io/platforms/go/guides/http/).

This PR makes it no that hub's scope is used when adding breadcrumbs.

This also removes the need to manually create a scope to use the breadcrumbs feature.

TheZeroSlave commented:
hi!
the sentryScope was introduced because we don't want to collect all breadscrumbs into the one global hub scope by default. it seems like a memory leak by design.
what we really want to have - isolated scope per request. if you don't want to miss the info, may be rewrite PR and add new method like zapsentry.NewScope() but passing already existing scope?

jeandeaual committed:
Add a function to add a new scope field from an existing Sentry scope

This allows to use breadcrumbs while keeping the information tied to
non-default hubs by doing `log.With(zapsentry.NewScopeFromScope(hub.Scope()))`
(e.g. when using https: //docs.sentry.io/platforms/go/guides/http/).

jeandeaual commented:
> hi! the sentryScope was introduced because we don't want to collect all breadscrumbs into the one global hub scope by default. it seems like a memory leak by design. what we really want to have - isolated scope per request. if you don't want to miss the info, may be rewrite PR and add new method like zapsentry.NewScope() but passing already existing scope?

@TheZeroSlave I see, thanks for the explanation.

For my use-case, I want to collect all breadcrumbs in the scope of the hubs created by the Sentry HTTP middleware, but I see how changing the way breadcrumbs are currently collected could be problematic.

I implemented your suggestion instead by adding a `NewScopeFromScope` method. This seems to be working fine for me!

jeandeaual commented:
By the way, I noticed that the example from the README doesn't work for me (the breadcrumbs are not being collected):

```go
core, err := zapsentry.NewCore(cfg, zapsentry.NewSentryClientFromClient(client))
	
// to use breadcrumbs feature - create new scope explicitly
log = log.With(zapsentry.NewScope())
	
// in case of err it will return noop core. so we can safely attach it
if err != nil {
	log.Warn("failed to init zap", zap.Error(err))
}
return zapsentry.AttachCoreToLogger(core, log)
```

Instead, I had to call `log.With(zapsentry.NewScope())` after `zapsentry.AttachCoreToLogger(core, log)`, otherwise the scope wouldn't be picked up:

```go
core, err := zapsentry.NewCore(cfg, zapsentry.NewSentryClientFromDSN(DSN))

// in case of err it will return noop core. so we can safely attach it
if err != nil {
	log.Warn("failed to init zap", zap.Error(err))
}

log = zapsentry.AttachCoreToLogger(core, log)

// to use breadcrumbs feature - create new scope explicitly
log = log.With(zapsentry.NewScope())

return log
```

Is this intended behavior?

TheZeroSlave commented:
> Is this intended behavior?

well, i tried it locally and didn't managed to reproduce this. The created scope passed to 
```
_ = c.client.CaptureEvent(event, nil, c.scope())
```
in both cases. Did you find a root cause of the problem?
